awscli/clidriver.py

```diff
@@ -145,10 +140,47 @@ def no_pager_handler(session, parsed_args, **kwargs):
     if parsed_args.no_cli_pager:
         config_store = session.get_component('config_store')
         config_store.set_config_provider(
-            'pager',  ConstantProvider(value=None)
+            'pager', ConstantProvider(value=None)
         )
 
 
+class AWSCLIEntryPoint:
+    def __init__(self):
+        self._error_handler = construct_entry_point_handlers_chain()
+
+    def main(self, args):
+        try:
+            rc = self._do_main(args)
+        except BaseException as e:
+            LOG.debug("Exception caught in AWSCLIEntryPoint", exc_info=True)
+            return self._error_handler.handle_exception(
+                e,
+                stdout=get_stdout_text_writer(),
+                stderr=get_stderr_text_writer()
+            )
+
+        HISTORY_RECORDER.record('CLI_RC', rc, 'CLI')
+        return rc
+
+    def _do_main(self, args):
+        driver = create_clidriver()
+        autoprompt_driver = AutoPromptDriver(driver)
+        auto_prompt_mode = autoprompt_driver.resolve_mode(args)
+        if auto_prompt_mode == 'on':
+            args = autoprompt_driver.prompt_for_args(args)
+            rc = driver.main(args)
+        elif auto_prompt_mode == 'on-partial':
+            autoprompt_driver.inject_silent_param_error_handler(driver)
```

Comment:
Let's rename this method to: `inject_silence_param_error_msg_handler(driver)` and the corresponding handler to `SilenceParamValidationMsgErrorHandler`. Mainly when I read silence, I was expecting it to swallow the entire exception and return a rc of 0. Adding the `Msg` part and switching to `Silence` better indicates that we are just not printing out the message.